### PR TITLE
DAOS-9855 test: Major refactoring of pool/evict.py

### PR DESCRIPTION
**Pull request by shimizukko**
- The code in `test_evict` was broken and also difficult to understand, so rewrote the whole test case while keeping the test steps the same.
- Removed `test_evict_bad_uuid`. It calls dmg pool evict with invalid UUID and it causes an error, but that's the general invalid argument error and not related to the evict functionality. Also, it's covered by the UUID corner case test.
- Removed the use of `pool_handle_exist()` and `test_param` parameter as this check is irrelevant because we use dmg.
- Enabled code block commented as DAOS-3836.

Skip-unit-tests: true
Test-tag: pool_evict_basic pool_evict_bad_uuid
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/tests/ftest/pool/evict.py [hunk 1]
```diff
@@ -56,204 +53,160 @@ def connected_pool(self, hostlist, targets=None):
...
         """
-        # setup pool and connect
-        self.pool = self.connected_pool(self.hostlist_servers)
+        host_list = []
 
-        self.log.info(
-            "Pool UUID: %s\n Pool handle: %s\n",
-            self.pool.uuid, self.pool.pool.handle.value,)
+        for rank in ranks:
+            host_list.append(self.server_managers[0].get_host(rank=rank))
```
Comment (daltonbohning) [inline src/tests/ftest/pool/evict.py:72]: Optional, if you want a one-liner :)
```
return [self.server_managers[0].get_host(rank=rank) for rank in ranks]
```

File: src/tests/ftest/pool/evict.py [hunk 2]
```diff
@@ -56,204 +53,160 @@ def connected_pool(self, hostlist, targets=None):
...
+        9. For the first two pools, check the UUID obtained from the pool_query() in
+        the previous step and verify that the evict on the third pool didn't have ill
+        effect.
 
         :avocado: tags=all,pr,daily_regression
-        :avocado: tags=small
-        :avocado: tags=pool,pool_evict,pool_evict_basic
+        :avocado: tags=vm
+        :avocado: tags=pool
+        :avocado: tags=pool_evict,pool_evict_basic
```
Comment (daltonbohning) [inline src/tests/ftest/pool/evict.py:112]: FYI, these tags are more inline with the comment I made on the pool destroy PR. I.e. if we use `pool_destroy`, it's more inline with `pool_evict` here.

File: src/tests/ftest/pool/evict.py [hunk 3]
```diff
@@ -56,204 +53,160 @@ def connected_pool(self, hostlist, targets=None):
...
-            # should fail the pool query because the handle was removed
+        pools = []
+        containers = []
+
+        # four_ranks stores the four ranks we have.
+        four_ranks = [0, 1, 2, 3]
+        # two_ranks_with is the two ranks that we'll create the third pool on.
+        two_ranks_with = [0, 1]
+        # two_ranks_without is the two ranks that we will not create a pool.
+        two_ranks_without = [2, 3]
```
Comment (daltonbohning) [inline src/tests/ftest/pool/evict.py:124]: Actually, I think the previous way of using "all" and "half" is better than hardcoding "four" and "two". If we ever want to run this test on a smaller or larger cluster, the test would need to be updated. Is it easy to keep using "all" and "half"?

**Review by daltonbohning — APPROVED** [review]
LGTM. Thanks for the updates!

File: src/tests/ftest/pool/evict.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/pool/evict.py]: Done. Thanks

File: src/tests/ftest/pool/evict.py (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/pool/evict.py]: Done. Thanks

File: src/tests/ftest/pool/evict.py (hunk 3 above)
Comment (shimizukko) [reply src/tests/ftest/pool/evict.py]: Thanks for the comment. I noticed that it's easy to support arbitrary number of hosts, so updated the code.
